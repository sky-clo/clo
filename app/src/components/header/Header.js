import React from "react";
import { useState } from "react";

import { ReactComponent as Logo } from "../../images/logo.svg";
import { ReactComponent as MenuIcon } from "../../images/menu.svg";
import { ReactComponent as ExitIcon } from "../../images/exit.svg";
import styles from "./header.module.scss";

export default function Header() {
  const [isMenuOpen, setIsMenuOpen] = useState(false);
  return (
    <>
      <header>
        <a className={styles.logoContainer} href="/">
          <Logo />
        </a>

        {/* Mobile hamburger icon and menu */}
        <button
          onClick={() => setIsMenuOpen(true)}
          className={styles.menuButton}
        >
          <MenuIcon
            className={styles.icon}
            onClick={() => setIsMenuOpen(true)}
          />
        </button>

        {/* Desktop inline header buttons */}
        <div className={styles.authButtons}>
          <a href="/" className="c-btn c-btn--secondary u-margin-right">
            Sign In
          </a>
          <a href="/" className="c-btn c-btn--primary u-margin-right">
            Create an Account
          </a>
        </div>
      </header>

      <dialog className={styles.menu} aria-hidden={!isMenuOpen}>
        <button onClick={() => setIsMenuOpen(false)}>
          <ExitIcon className={styles.icon} />
        </button>

        <div className={styles.menuItems}>
          <a href="/" className="c-btn c-btn--secondary u-margin-right">
            Sign In
          </a>
          <a href="/" className="c-btn c-btn--primary u-margin-right">
            Create an Account
          </a>
        </div>
      </dialog>
    </>
  );
}
