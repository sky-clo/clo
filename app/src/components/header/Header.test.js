import React from "react";
import { shallow } from "enzyme";

import Header from "./Header";

describe("Header", () => {
  it("should match snapshot", () => {
    const render = shallow(<Header />);
    expect(render).toMatchSnapshot();
  });
});
