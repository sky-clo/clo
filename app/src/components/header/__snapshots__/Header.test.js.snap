// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header should match snapshot 1`] = `
<Fragment>
  <header>
    <a
      className="logoContainer"
      href="/"
    >
      <ForwardRef(SvgLogo) />
    </a>
    <button
      className="menuButton"
      onClick={[Function]}
    >
      <ForwardRef(SvgMenu)
        className="icon"
        onClick={[Function]}
      />
    </button>
    <div
      className="authButtons"
    >
      <LinkButton
        data-test="Header-sign-in"
        href="/sign-in"
        primary={false}
      >
        Sign In
      </LinkButton>
      <LinkButton
        data-test="Header-create-an-account"
        href="/create-an-account"
        primary={true}
      >
        Create an Account
      </LinkButton>
    </div>
  </header>
  <dialog
    aria-hidden={true}
    className="menu"
  >
    <button
      onClick={[Function]}
    >
      <ForwardRef(SvgExit)
        className="icon"
      />
    </button>
    <div
      className="menuItems"
    >
      <LinkButton
        data-test="Header-sign-in"
        href="/sign-in"
        primary={false}
      >
        Sign In
      </LinkButton>
      <LinkButton
        data-test="Header-create-an-account"
        href="/create-an-account"
        primary={true}
      >
        Create an Account
      </LinkButton>
    </div>
  </dialog>
</Fragment>
`;
